### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by scargill** [csuermann/node-red-contrib-virtual-smart-home#188]
The node is stuck in INITIALISING.
I made a "virtual light" which is a SWITCH. I have my Amazon account linked to virtual smart home - Node-Red is up to date and working perfectly.

I've filled in as follows: 
![image](https://user-images.githubusercontent.com/1889411/221297109-47256d93-04e6-4fe3-b9f3-218b49e2e454.png)

Can't get past this point. Any ideas?

**Comment by csuermann** [comment]
Hi @scargill,

Thanks for reaching out! 

I'm guessing this might be a network issue. You could try to switch on the "Debug" option for your vsh-connection node and inspect what's being logged to the Node-RED CLI.

Also, which version of Node.js, Node-RED and VSH are you using?

**Comment by scargill** [comment]
Hi there

Ok, easy bits first... 
node -v returnd v14.21.1
node-red v2.1.3

(in node-red startup)
25 Feb 17:22:28 - [info] [vsh-connection:maureen@scargill.org] MQTT disconnected

In the node - which says "initializing" all the time - I have "connection" which is:

maureen@scargill.org - and Thing ID:  vsht-35299d510425e0941f-0fcxxxxxxxxx (xxxxxxxxx obviously hex as with the rest).
Debug is ticked... current plan is free.

If I go into Node-red aat the CLI, soon after startup I see:

25 Feb 17:29:54 - [info] [vsh-connection:maureen@scargill.org] Attempting MQTT connection: a1pv0eq8s016ut-ats.iot.eu-west-1.amazonaws.com:8883 (clientId: vsht-35299d51-81e1-425e-941f-0fcxxxxxxxxx)
25 Feb 17:29:54 - [info] [vsh-connection:maureen@scargill.org] MQTT subscribe to topics: ["$aws/things/vsht-35299d51-81e1-425e-941f-0fcxxxxxxxxx/shadow/get/accepted","vsh/vsht-35299d51-81e1-425e-941f-0fcxxxxxxx/+/directive","vsh/service","vsh/version/3.0.9/+","vsh/vsht-35299d51-81e1-425e-941f-0fcxxxxxxxxx/service"]
25 Feb 17:29:54 - [info] [vsh-connection:maureen@scargill.org] MQTT disconnected
25 Feb 17:29:54 - [info] [vsh-connection:maureen@scargill.org] MQTT disconnected
Error: Connection closed

My alexas work (I have an alexa in this room and an echo show in the living room)

Does that help?

**Comment by csuermann** [comment]
Thanks for the additional information! I have some more questions ;-)

1. Did it work before you updated to the latest version of VSH?
2. Did you create your Node-RED instance from a backup or imported your flows from another machine?
3. Is there anything in your setup blocking port 8883?
4. Have you tried deleting your vsh-connection and create a new one?

**Comment by scargill** [comment]
Hi

I don't think it worked but its a long time since I tried it. Oh no that 
message was there from a previous attempt.

I always use the same node red.. I use rpi clone regularly to clone the 
setup on my rpi 4 and if things don't work out I can always go back.

Not using 8883 for anything as far as I know. I have a glinet 4g router 
pulling in 4g and feeding that to another normal glinet router.. hmm that 
might have adguardpro on it. I could try turning that off.

Quite happy to try anything except starting from scratch.

Pete

**Comment by csuermann** [comment]
I could not find your Thing ID in the VSH backend database, so I guess it has been deleted due to inactivity in accordance with the [terms of use](https://github.com/csuermann/node-red-contrib-virtual-smart-home#note).

You should be able to bring everything back to life by deleting the stale vsh-connection from within Node-RED, then create a new one and go through the account linking process again.

Hope that helps!

**Comment by scargill** [comment]
So -- ~I dleted all devices connected to the connection and dleted the config node in NR.

I went to create a new hsh connection, called it mylamp, the old connection was indeed no longer available created a new one at which point the ACCOUNDT linking box popped up, I went to https://amazon.com/gb/code  and filled in the code - 
![image](https://user-images.githubusercontent.com/1889411/221404897-6131e2c9-75fb-45f4-a287-e4b0e69276d8.png)
then hit continue. all well.. the details appeared after a few seconds as expected and I hit ADD... . gave the name mylamp,. topic mylamp - entered "color changing light bulb" option, didnt tick the other boxes and hit DONE - at that point mylamp had a red triangle on it . - I hit DEPLOY ..

The workspace contains some nodes that are not properly configured:

[Amazon Alexa] mylamp (vsh-virtual-device)
Are you sure you want to deploy?

Am I missing something?
